tools/generator/filters/caml.py

```diff
@@ -0,0 +1,87 @@
+import textwrap
+
+from . import register_filter
+from .common import is_array
+
+
+@register_filter
+def caml_type(value: str) -> str:
+    if value == 'void':
+        return 'unit'
+    if value == 'double':
+        return 'float'
+    if is_array(value):
+        return '{} array'.format(value)
+    return value
+
+
+@register_filter
+def caml_callback(func) -> str:
+    '''symbolic name of a function to be used with Callback.register in caml
+    code, and with caml_named_value in C code
+    '''
+    return 'ml_' + func['fct_name']
+
+
+# caml functions need to take at least one parameter, be it a unit value if
+# there is nothing useful to give
+
+@register_filter
+def cxx_nonempty_or_unit(arg_list):
```

Comment:
Why `cxx_`?

Reply:
It's used in parameters list in C++ functions, see https://github.com/prologin/stechec2/pull/97/files#diff-0ab55a5f02d7d1a0896ff290cce05feeR177.

Reply:
Okay, but this still gets added to the global scope available to all languages, so you should at least mention ocaml in the name